locals.tf

```diff
@@ -2,15 +2,33 @@ locals {
   # Properly format usernames for use in an ARN
   iam_usernames = contains(var.iam_usernames, "root") ? ["root"] : formatlist("user/%s", var.iam_usernames)
 
+  # A list of object actions that allow for reading.
+  object_actions_read = ["s3:GetObject*"]
+
+  # A list of object permissions that allow for writing.
+  object_actions_write = [
+    "s3:DeleteObject",
+    "s3:PutObject*",
+  ]
+
+  # A list of object permissions that allow for both reading and
+  # writing.
+  object_actions_read_write = concat(
+    local.object_actions_read,
+    local.object_actions_write,
+  )
+
   # If var.role_description contains two instances of "%s", use format()
   # to replace the first "%s" with var.s3_bucket and the second "%s"
   # with var.entity_name, otherwise just use var.role_description as is
   role_description = length(regexall(".*%s.*%s.*", var.role_description)) > 0 ? format(var.role_description, var.s3_bucket, var.entity_name) : var.role_description
```

Comment:
I think the role description should be expanded to include a third value replacement of either "read-only" or "read-write", similar to what you did in https://github.com/cisagov/s3-read-role-tf-module/pull/47/commits/f95171913cb01e844d2afa2e1bb182a07a4bc2a6.

Even though the role name will include "ReadOnly" or "ReadWrite", I think it's preferable to document that info in the description as well.

Reply:
Does commit 137746d do it for you?